src/librustc_mir/borrow_check/mod.rs

```diff
@@ -740,6 +740,30 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                     self.visit_terminator_drop(loc, term, flow_state, &place, field_ty, span);
                 }
             }
+            // Same as above, but for tuples.
+            ty::TyTuple(tys) => {
```

Comment:
Hmm, maybe we can extract a helper function here? If feels a bit silly to have basically the same code repeated so many times.

Reply:
Done. I preferred a closure to avoid having to pass all of the captured variables as arguments.